Packet++/src/FtpLayer.cpp

```diff
@@ -0,0 +1,396 @@
+#define LOG_MODULE PacketLogModuleFtpLayer
+
+#include "FtpLayer.h"
+
+namespace pcpp
+{
+
+	// ----------------- Class FtpRequestLayer -----------------
+	FtpRequestLayer::FtpRequestLayer()
+	{
+		m_Protocol = FTP;
+		m_Data = NULL;
+		m_DataLen = 0;
+	}
+
+	void FtpRequestLayer::setCommand(FtpCommand code)
+	{
+		setCommandField(getCommandAsString(code));
+	}
+
+	FtpRequestLayer::FtpCommand FtpRequestLayer::getCommand() const
+	{
+		size_t val = 0;
+		std::string field = getCommandField();
+
+		for (size_t idx = 0; idx < field.size(); ++idx)
+			val |= (field.c_str()[idx] << (idx * 8));
+
+		return static_cast<FtpCommand>(val);
```

Comment:
there is a question of what would happen if the command in the packet is an invalid command, for example: `XXXX`. Usually it's a good practice to check the value before using `static_cast`, but here it would be very difficult. I'm not sure I have a good suggestion here...

Reply:
@seladb Looks like for C+11 there is no problem (no Undefined Behaviour) if I understand correctly (see accepted answer [here](https://stackoverflow.com/questions/18195312/what-happens-if-you-static-cast-invalid-value-to-enum-class)) but for ensure safety maybe we can use an underlying type for enum (discussed [here](https://stackoverflow.com/questions/55392083/c-enum-class-cast-to-non-existing-entry)). Since the commands have at most four characters `int` is sufficient. What do you think?

For the user side, if anyone is parsing a non-standard command value, the user should convert it to integer and compare it with its custom values.

Reply:
Yes that's a good idea! we can convert the enum to `enum class FtpCommand : int`, and add an `Unknown` value as the default value. Then casting will be safe,

We can probably do the same for `FtpStatusCode`